msi-ec.c

```diff
@@ -2381,6 +2381,83 @@ static struct msi_ec_conf CONF28 __initdata = {
 	},
 };
 
+static const char *ALLOWED_FW_29[] __initconst = {
+	"16Q4EMS1.110", // GS65 Stealth
+	NULL
+};
+
+static struct msi_ec_conf CONF29 __initdata = {
+	.allowed_fw = ALLOWED_FW_29,
+	.charge_control = {
+		.address      = 0xef,
+		.offset_start = 0x8a,
+		.offset_end   = 0x80,
+		.range_min    = 0x8a,
+		.range_max    = 0xe4,
+	},
+	.webcam = {
+		.address       = 0x2e,
+		.block_address = MSI_EC_ADDR_UNSUPP,
+		.bit           = 1,
+	},
+	.fn_win_swap = {
+		.address = 0xbf,
+		.bit     = 4, // 0x00-0x10
+	},
+	.cooler_boost = {
+		.address = 0x98,
+		.bit     = 7,
+	},
+	.shift_mode = {
+		.address = 0xf2,
+		.modes = {
+			{ SM_ECO_NAME,     0xc2 }, // super battery
+			{ SM_COMFORT_NAME, 0xc1 }, // balanced
+			{ SM_TURBO_NAME,   0xc4 }, // extreme
+			{ SM_SPORT_NAME,   0xc0 }, // sport
+			MSI_EC_MODE_NULL
+		},
+	},
+	.super_battery = {
+		.address = MSI_EC_ADDR_UNSUPP, // Function not shown in dragon center
+	},
```

Comment:
> // Function not shown in dragon center

Are you sure? You added super battery to the shift mode list

Reply:
Is it the eco mode ?
If true, what am i supposed to write at the address of the super battery ?

Reply:
Can you screenshot which modes you have?

Reply:
![Uploading image.png…]()

Reply:
![battery](https://github.com/user-attachments/assets/49606d13-e597-43c1-b97a-b0580d0cad44)
![modes](https://github.com/user-attachments/assets/86a71782-1234-4ab2-8365-92e9c7266f33)

Reply:
Here is all I have

Reply:
Can you screenshot available fan modes? Probably in turbo or sport mode

Reply:
![Screenshot 2024-10-08 205921](https://github.com/user-attachments/assets/6c624997-4be2-41cc-8650-84c11989078c)